internal/bsr/bsr.go

```diff
@@ -200,8 +200,11 @@ func OpenSession(ctx context.Context, sessionRecordingId string, f storage.FS, k
 	// Load and verify recording metadata
 	sha256Reader, err := crypto.NewSha256SumReader(ctx, cc.metaFile)
 	if err != nil {
+		cc.metaFile.Close()
 		return nil, fmt.Errorf("%s: %w", op, err)
 	}
+	defer sha256Reader.Close()
```

Comment:
I think this Close() function will return an error and we will lose that context with the defer. I think that we should create an anonymous function for this defer, like so:
```
defer func() {
if closeErr := sha256Reader.Close(); closeErr != nil {
    // wrap or join closeErr with err
}
}()
```
In order to keep the context of returning the closeErr from the anonymous function, we will need to utilize named return values.